library/Vanilla/Embeds/TwitchEmbed.php

```diff
@@ -0,0 +1,209 @@
+<?php
+/**
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license https://opensource.org/licenses/GPL-2.0 GPL-2.0
+ */
+
+namespace Vanilla\Embeds;
+
+use Exception;
+
+/**
+ * Class for parsing twitch URLs, fetching twitch OEmbed data, and rendering server side HTML for a twitch video embed
+ */
+class TwitchEmbed extends VideoEmbed {
+
+    const DEFAULT_HEIGHT = 300;
+
+    const DEFAULT_WIDTH = 400;
+
+    /** @inheritdoc */
+    protected $domains = ['www.twitch.tv', 'clips.twitch.tv'];
+
+    /** @var string */
+    private $urlType;
+
+    public function __construct()
+    {
+        parent::__construct('twitch', 'video');
+        $this->urlType = '';
```

Comment:
Why set a default here, instead of defining the property when you're declaring it?